gsa/src/web/components/form/useForm.js

```diff
@@ -0,0 +1,109 @@
+/* Copyright (C) 2019 Greenbone Networks GmbH
+ *
+ * SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+import {useState, useEffect} from 'react';
+
+export const capitalizeFirstLetter = string => {
+  if (string.length > 0) {
+    return string[0].toUpperCase() + string.slice(1);
+  }
+  return '';
+};
+
+export const parseAlias = string => {
+  const parsedString = string.replace(/(_)/g, ' ').replace(/([A-Z])/g, ' $1');
+
+  const words = parsedString.split(' ');
+  let errorString = '';
+
+  words.forEach(word => (errorString += capitalizeFirstLetter(word) + ' '));
+  return errorString.trim(); // parse "name" to something human readable. Accepts underscores and camelcase. "credential_id" > "Credential Id", etc.
+};
+
+export const syncVariables = (values, formState, dependencies = {}) => {
+  Object.keys(formState).forEach(key => (values[key] = formState[key]));
+  Object.keys(dependencies).forEach(key => (values[key] = dependencies[key]));
+}; // sync all form states with the values known to SaveDialog
+
+const useForm = (stateSchema, validationRules, callback, extras = {}) => {
```

Comment:
I think, we should name this more appropriately. We are not using a form, we are using form validation. => I'd call it useFormValidation.

Reply:
This is a great idea. Implemented.